rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -129,6 +135,15 @@ local function sync_backend(backend)
 end
 
 local function sync_backends()
+  local timestamp = configuration.get_timestamp_data()
+  if timestamp <= last_timestamp then
+    for _, external_backend in pairs(external_backends) do
+      sync_backend(external_backend)
+    end
+    return
+  end
+  last_timestamp = timestamp
```

Comment:
Are you deliberately updating this in the beginning of function body to avoid retries in case of i.e json decoding error happens? Intuitively I'd have expected this update to be in the end of the function body. I wonder how we can make the intention clearer for the reader.